src/sdl/sdl_joystick.c

```diff
@@ -97,7 +94,7 @@ void _al_sdl_joystick_event(SDL_Event *e)
 static bool sdl_init_joystick(void)
 {
    count = SDL_NumJoysticks();
-   joysticks = calloc(count, sizeof * joysticks);
+   joysticks = count > 0 ? calloc(count, sizeof * joysticks) : NULL;
```

Comment:
In my testing I got errors when going from 1 -> 0 controllers because although `joysticks` was being freed in `sdl_exit_joystick`, this line here would have a count of zero and I noticed `joysticks[0]` still had a pointer to (free'd) data. [`calloc`](https://en.cppreference.com/w/c/memory/calloc) has odd behavior when size is zero. To fix, manually set to NULL when size is zero.

Also, set `joysticks` to NULL after freeing for good measure.